### ISSUE TITLE
Domain adaptation  fine tune VS instruction_tuned 

### ISSUE THREAD

**Issue by ronyadgar** [huggingface/accelerate#2130]
TL;DR: I want to fine-tune 'meta-llama/Llama-2-7b-chat-hf' with my own instructional data, but it turns out that the new model is worse than the original one.

How should I differentiate in the code between different types of fine-tuning? I saw different types of fine-tuning on the internet:

Domain adaption
Instruction-based fine-tuning (it looks like this is what is needed)
Supervised fine-tuning ...
But how should it be different in the code? For example, assuming that I use AutoModelForCausalLM and a trainer object, a simple code would look like this:

```python

model = AutoModelForCausalLM.from_pretrained("meta-llama/Llama-2-7b-chat-hf")
    trainer = Trainer(

        model=model,
        args=training_args,
        train_dataset=dataset_train,
        eval_dataset=dataset_test,
        data_collator=transformers.DataCollatorForLanguageModeling(
            tokenizer, mlm=False
        )
```        
      
But how dataset_train should look like ?

**Comment by muellerzr** [comment]
The github is reserved for bug reports, please direct your post to https://discuss.huggingface.co :)

**Comment by dpalacioj** [comment]
Hello, I have the same problem. Until now I haven't known how to solve it. Were you able to improve your results and understand how train_data is different in finetuning as an instruction?
